### ISSUE TITLE
`excludeGuardClauses` in `ReturnCount` should ignore first `super` call

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#5680]
## Expected Behavior of the rule
When `excludeGuardClauses` flag is true for `ReturnCount` it should ignore the first `super.` call of the function call. I frequently encounter this when writing visitor in detekt rules.

One example where this rule flags a code smell is as below:
```
override fun visitWhenExpression(expression: KtWhenExpression) {
        super.visitWhenExpression(expression)
        if (expression.entries.size == 2) return
        if (expression.elseExpression != null) return
        if (ignoreWhenContainingVariableDeclaration.not() && expression.subjectExpression is KtProperty) return
        if (expression.entries.none { it.conditions.size > 1 }) return
        report(
            CodeSmell(
                issue,
                Entity.from(expression),
                "Prefer using 'if' for binary conditions instead of 'when'."
            )
        )
    }
```

As per me, lines `if (expression.entries.size == 2) return`, `if (expression.elseExpression != null) return` and such are acting as a guard clause. But just because they are below `super` they are getting calculated for the `ReturnCount` threshold calculation

Failing TC that should pass
```kotlin
@Test
fun `does not report when guard clause is present below super call`() {
    val code = """
    open class A {
        open fun a(list: List<Int>) {
            
        }
    }
    
    class B: A() {
        override fun a(list: List<Int>) {
            super.a(list)
            
            if (list.isEmpty()) return
            if (list.contains(-1)) return
            if (list.contains(-2)) return
            if (list.contains(-3)) return
        }
    }
""".trimIndent()
    val findings = ReturnCount(
        TestConfig(
            mapOf(
                EXCLUDE_GUARD_CLAUSES to "true",
            )
        )
    ).compileAndLint(code)
    assertThat(findings).isEmpty()
}
```

**Comment by cortinico** [comment]
> When `excludeGuardClauses` flag is true for `ReturnCount` it should ignore the first `super.` call of the function call. I frequently encounter this when writing visitor in detekt rules.

Totally agree. This should be doable by tweaking the `yieldStatementsSkippingGuardClauses` util function we have in the codebase

**Comment by atulgpt** [comment]
Hi @cortinico glad you liked the idea @BraisGabin what are your thoughts on this?
If you both agree that this is something we should do, then I can pick this up

**Comment by BraisGabin** [comment]
Sure, this have a lot of sense.

**Comment by cortinico** [comment]
Assigned to you @atulgpt 👍
